gnomad/utils/vep.py

```diff
@@ -121,17 +148,40 @@ def vep_or_lookup_vep(ht, reference_vep_ht=None, reference=None, vep_config=None
                 f'vep_or_lookup_vep got {reference}. Expected one of {", ".join(possible_refs)}'
             )
 
-        reference_vep_ht = hl.read_table(vep_context_ht_path(reference))
+        vep_context = get_vep_context(reference)
+        if vep_version is None:
+            vep_version = vep_context.default_version
+
+        if vep_version not in vep_context.versions:
+            logger.warning(
+                f"No VEPed context Table available for genome build {reference} and VEP version {vep_version}, "
+                f"all variants will be VEPed using the following VEP:\n{vep_help}"
+            )
+            return hl.vep(ht, vep_config_path)
+
+        logger.info(
+            f"Using VEPed context Table from genome build {reference} and VEP version {vep_version}"
+        )
+
+        reference_vep_ht = vep_context.versions[vep_version].ht()
+        vep_context_help = hl.eval(reference_vep_ht.vep_help)
+        vep_context_config = hl.eval(reference_vep_ht.vep_config)
+
+        assert vep_help == vep_context_help, (
```

Comment:
I wasn't going to because then is requires them to add those globals, but happy to force users to add them if you think it would be helpful

Reply:
I could go either way. It would be nice to have the guard rails, but it's not obvious how to make a Table compatible with these functions. No objection to leaving them out, though it might be worth a note in the docstring to make sure the `reference_vep_ht` was generated with the same version of VEP / VEP configuration.